### ISSUE TITLE
[v3] `treeshake` unexpectedly assumed `"a" in b`==`b.a != null`

### ISSUE THREAD

**Issue by liuxingbaoyu** [rollup/rollup#4680]
### Rollup Version

3.2.2

### Operating System (or Browser)

all

### Node Version (if applicable)

14-18

### Link To Reproduction

[repl](https://rollupjs.org/repl/?version=3.2.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiolMjBUUkVFLVNIQUtJTkclMjAqJTJGJTVDbmltcG9ydCUyMColMjBhcyUyMHZpcnR1YWxUeXBlcyUyMGZyb20lMjAlNUMlMjIuJTJGdmlydHVhbC10eXBlcyU1QyUyMiUzQiU1Q24lNUNuZnVuY3Rpb24lMjBpc1ZpcnR1YWxUeXBlKHR5cGUpJTIwJTdCJTVDbiUyMCUyMHJldHVybiUyMHR5cGUlMjBpbiUyMHZpcnR1YWxUeXBlcyUzQiU1Q24lN0QlNUNuJTVDbmZ1bmN0aW9uJTIwZXhwbG9kZSh0eXBlKSUyMCU3QiU1Q24lMjAlMjBpZighKHR5cGUlMjBpbiUyMHZpcnR1YWxUeXBlcykpcmV0dXJuJTNCJTVDbiU1Q24lMjAlMjBjb25zdCUyMHR5cGVzJTIwJTNEJTIwdmlydHVhbFR5cGVzJTVCdHlwZSU1RCU1Q24lNUNuJTIwJTIwaWYodmlydHVhbFR5cGVzJTVCdHlwZSU1RCUyMCElM0QlMjBudWxsKSU3QiU1Q24lMjAlMjAlMjAlMjAlMkYlMkYlMjBoZXJlJTIwJUUyJTg2JTkxJTVDbiUyMCUyMCUyMCUyMGZvciUyMChjb25zdCUyMHR5cGUlMjBvZiUyMHR5cGVzKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjBpZiUyMCh2aXNpdG9yJTVCdHlwZSU1RCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwbWVyZ2VQYWlyKHZpc2l0b3IlNUJ0eXBlJTVEJTJDJTIwZm5zKSUzQiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlN0QlNUNuJTIwJTIwJTIwJTIwJTdEJTVDbiUyMCUyMCU3RCU1Q24lN0QlNUNuJTVDbnZhciUyMHZpc2l0b3JzJTIwJTNEJTIwJTJGKiUyM19fUFVSRV9fKiUyRk9iamVjdC5mcmVlemUoJTdCJTVDbiUyMCUyMF9fcHJvdG9fXyUzQSUyMG51bGwlMkMlNUNuJTIwJTIwZXhwbG9kZSUzQSUyMGV4cGxvZGUlMkMlNUNuJTIwJTIwdmVyaWZ5JTNBJTIwdmVyaWZ5JTJDJTVDbiUyMCUyMG1lcmdlJTNBJTIwbWVyZ2UlNUNuJTdEKSUzQiU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJ2aXJ0dWFsLXR5cGVzLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGNvbnN0JTIwYSUzRCU1QiU1RCUzQiU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

### Expected Behaviour

```js
/* TREE-SHAKING */
import * as virtualTypes from "./virtual-types";

function isVirtualType(type) {
  return type in virtualTypes;
}

function explode(type) {
  if(!(type in virtualTypes))return;

  const types = virtualTypes[type]

  if(virtualTypes[type] != null){
    // here ↑
    for (const type of types) {
      if (visitor[type]) {
        mergePair(visitor[type], fns);
      }
    }
  }
}

var visitors = /*#__PURE__*/Object.freeze({
  __proto__: null,
  explode: explode,
  verify: verify,
  merge: merge
});

```

### Actual Behaviour

```js
'use strict';

const a=[];

var virtualTypes = /*#__PURE__*/Object.freeze({
  __proto__: null,
  a: a
});

/* TREE-SHAKING */

function explode(type) {
  if(!(type in virtualTypes))return;

  const types = virtualTypes[type];

  {
    // here ↑
    for (const type of types) {
      if (visitor[type]) {
        mergePair(visitor[type], fns);
      }
    }
  }
}

/*#__PURE__*/Object.freeze({
  __proto__: null,
  explode: explode,
  verify: verify,
  merge: merge
});

```

**Comment by lukastaegert** [comment]
Thanks! Simplified example: https://rollupjs.org/repl/?version=3.2.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHZpcnR1YWxUeXBlcyUyMGZyb20lMjAlNUMlMjIuJTJGdmlydHVhbC10eXBlcyU1QyUyMiUzQiU1Q24lNUNuaWYlMjAodmlydHVhbFR5cGVzJTVCZ2xvYmFsVGhpcy51bmtub3duJTVEJTIwISUzRCUzRCUyMG51bGwpJTIwJTdCJTVDbiUyMCUyMGxvZyglNUMlMjJmYWlsJTVDJTIyKSU1Q24lN0QlMjAlNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIydmlydHVhbC10eXBlcy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGElMjAlM0QlMjBudWxsJTNCJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

**Comment by lukastaegert** [comment]
Fix at #4684

https://rollupjs.org/repl/?pr=4684&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMHZpcnR1YWxUeXBlcyUyMGZyb20lMjAlNUMlMjIuJTJGdmlydHVhbC10eXBlcyU1QyUyMiUzQiU1Q24lNUNuaWYlMjAodmlydHVhbFR5cGVzJTVCZ2xvYmFsVGhpcy51bmtub3duJTVEJTIwISUzRCUzRCUyMG51bGwpJTIwJTdCJTVDbiUyMCUyMGxvZyglNUMlMjJmYWlsJTVDJTIyKSU1Q24lN0QlMjAlNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIydmlydHVhbC10eXBlcy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGElMjAlM0QlMjBudWxsJTNCJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJjanMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

**Comment by liuxingbaoyu** [comment]
Thanks!
